src/components/Tour/Tour.jsx

```diff
@@ -0,0 +1,111 @@
+import './Tour.less';
+import * as d3 from 'd3'
+import React, { useMemo, useRef } from 'react';
+import { useDispatch, useSelector } from 'react-redux'
+import { selectQueryTab } from '../../reducers/query/selectors';
+import { selectViewIsTourEnabled } from '../../reducers/view/selectors';
+import { Steps } from 'intro.js-react';
+import { TOUR_STEPS } from './constants/tourStepsConstants';
+import { TourButton } from './TourButton';
+import { tourToggled } from '../../actions/trends';
+
+export const Tour = () => {
+  const dispatch = useDispatch();
+  const isTourEnabled = useSelector( selectViewIsTourEnabled );
+  const tab = useSelector( selectQueryTab );
+  const steps = useMemo( () =>
+     TOUR_STEPS[tab]
+  , [ TOUR_STEPS, tab ] );
+  const stepRef = useRef();
+
+    // INTRODUCTION / TUTORIAL OPTIONS:
+  const options = {
+    disableInteraction: true,
+    scrollToElement: true,
+    scrollTo: 'tooltip',
+    showStepNumbers: false,
+    exitOnOverlayClick: false,
+    exitOnEsc: true,
+    nextLabel: 'Next',
+    prevLabel: 'Previous',
+    doneLabel: 'End Tour',
+    steps: steps
+  };
+
+
+  /**
+   * Before Change handler
+   *
+   * @param {object} ref - React component reference.
+   */
+  function handleBeforeChange( ref ) {
+    const expand = ref.current.introJs.currentStep() + 1;
+
+    // hacks to setup the page. we should figure out if we should
+    // have this in redux instead
+    if ( expand === 9 ) {
```

Comment:
Should the `9` go into a constant?

Reply:
I updated this so it's more scalable so we don't have to hardcode the number.